src/Interpreters/addSubcolumnsExtraction.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <Interpreters/ActionsDAG.h>
+
+namespace DB
+{
+
+/// Create a new ActionsDAG where all input subcolumns that are not presented in the header are replaced to the getSubcolumn function.
+ActionsDAG addSubcolumnsExtraction(ActionsDAG dag, const Block & header, const ContextPtr & context);
```

Comment:
Maybe we can move `ActionsDAG::merge` outside and make a function like
`addSubcolumnsExtraction(const Block & available_columns, const Names & required_columns_or_subcolumns, context)` or something similar.

But this is minor.

Reply:
Changed